traitsui/tests/editors/test_button_editor.py

```diff
@@ -46,36 +49,25 @@ class ButtonTextEdit(HasTraits):
 )
 
 
-def get_button_text(button):
-    """ Return the button text given a button control """
-    if is_wx():
-        return button.GetLabel()
-
-    elif is_qt():
-        return button.text()
-
-
 @requires_toolkit([ToolkitName.qt, ToolkitName.wx])
-class TestButtonEditor(unittest.TestCase):
+class TestButtonEditor(unittest.TestCase, UnittestTools):
     def check_button_text_update(self, view):
         button_text_edit = ButtonTextEdit()
 
-        with reraise_exceptions(), \
-                create_ui(button_text_edit, dict(view=view)) as ui:
-
-            process_cascade_events()
-            editor, = ui.get_editors("play_button")
-            button = editor.control
-
-            self.assertEqual(get_button_text(button), "I'm a play button")
+        tester = UITester()
+        with tester.create_ui(button_text_edit, dict(view=view)) as ui:
+            button = tester.find_by_name(ui, "play_button")
+            actual = button.inspect(query.DisplayedText())
+            self.assertEqual(actual, "I'm a play button")
 
             button_text_edit.play_button_label = "New Label"
-            self.assertEqual(get_button_text(button), "New Label")
+            actual = button.inspect(query.DisplayedText())
+            self.assertEqual(actual, "New Label")
```

Comment:
The following tests are now failing on wx: 
```
======================================================================
ERROR: test_custom_button_editor (traitsui.tests.editors.test_button_editor.TestButtonEditor)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\appveyor\.edm\envs\traitsui-test-3.6-wx\lib\site-packages\traitsui\tests\editors\test_button_editor.py", line 77, in test_custom_button_editor
    self.check_button_text_update(custom_view)
  File "C:\Users\appveyor\.edm\envs\traitsui-test-3.6-wx\lib\site-packages\traitsui\tests\editors\test_button_editor.py", line 60, in check_button_text_update
    actual = button.inspect(query.DisplayedText())
  File "C:\Users\appveyor\.edm\envs\traitsui-test-3.6-wx\lib\site-packages\traitsui\testing\tester\ui_wrapper.py", line 129, in inspect
    return self._perform_or_inspect(interaction)
  File "C:\Users\appveyor\.edm\envs\traitsui-test-3.6-wx\lib\site-packages\traitsui\testing\tester\ui_wrapper.py", line 185, in _perform_or_inspect
    supported=supported,
traitsui.testing.tester.exceptions.InteractionNotSupported: No handler is found for target <class 'traitsui.wx.button_editor.CustomEditor'> with interaction <class 'traitsui.testing.tester.query.DisplayedText'>. Supported these: []
======================================================================
ERROR: test_custom_button_editor_clicked (traitsui.tests.editors.test_button_editor.TestButtonEditor)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\appveyor\.edm\envs\traitsui-test-3.6-wx\lib\site-packages\traitsui\tests\editors\test_button_editor.py", line 94, in test_custom_button_editor_clicked
    self.check_button_fired_event(custom_view)
  File "C:\Users\appveyor\.edm\envs\traitsui-test-3.6-wx\lib\site-packages\traitsui\tests\editors\test_button_editor.py", line 88, in check_button_fired_event
    button.perform(command.MouseClick())
  File "C:\Users\appveyor\.edm\envs\traitsui-test-3.6-wx\lib\site-packages\traitsui\testing\tester\ui_wrapper.py", line 116, in perform
    self._perform_or_inspect(interaction)
  File "C:\Users\appveyor\.edm\envs\traitsui-test-3.6-wx\lib\site-packages\traitsui\testing\tester\ui_wrapper.py", line 185, in _perform_or_inspect
    supported=supported,
traitsui.testing.tester.exceptions.InteractionNotSupported: No handler is found for target <class 'traitsui.wx.button_editor.CustomEditor'> with interaction <class 'traitsui.testing.tester.command.MouseClick'>. Supported these: []
======================================================================
ERROR: test_simple_button_editor (traitsui.tests.editors.test_button_editor.TestButtonEditor)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\appveyor\.edm\envs\traitsui-test-3.6-wx\lib\site-packages\traitsui\tests\editors\test_button_editor.py", line 74, in test_simple_button_editor
    self.check_button_text_update(simple_view)
  File "C:\Users\appveyor\.edm\envs\traitsui-test-3.6-wx\lib\site-packages\traitsui\tests\editors\test_button_editor.py", line 60, in check_button_text_update
    actual = button.inspect(query.DisplayedText())
  File "C:\Users\appveyor\.edm\envs\traitsui-test-3.6-wx\lib\site-packages\traitsui\testing\tester\ui_wrapper.py", line 129, in inspect
    return self._perform_or_inspect(interaction)
  File "C:\Users\appveyor\.edm\envs\traitsui-test-3.6-wx\lib\site-packages\traitsui\testing\tester\ui_wrapper.py", line 185, in _perform_or_inspect
    supported=supported,
traitsui.testing.tester.exceptions.InteractionNotSupported: No handler is found for target <class 'traitsui.wx.button_editor.SimpleEditor'> with interaction <class 'traitsui.testing.tester.query.DisplayedText'>. Supported these: [<class 'traitsui.testing.tester.command.MouseClick'>]
```
I think they are passing before. I can reproduce this locally.

They are failing on Appveyor and Travis too. However the wx build is allowed to fail, because there are still existing issues with wx we have not finished fixing. So I am afraid we cannot just rely on the green tick on the PR, we will need to check the CI job log for new failures and maybe run the test suite locally too.